src/Modal/Modal.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+export class Modal extends Component {
+  // select body element to add Modal component too
+  bodyElm = document.querySelector('body');
+
+  // send type of button (primary or secondary) text to onClose method
+  handleCloseClick = type => {
+    this.props.onClose(type);
+  };
+
+  // check for Escape key press
+  handleKeyPress = event => {
+    if (event.key === 'Escape' || event.key === 'Esc') {
+      this.handleCloseClick();
+    }
+  };
+
+  // add event listener for escape key
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyPress, false);
+  }
+
+  // remove event listener for escape key
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleKeyPress, false);
+  }
+
+  render() {
+    const { children, title, secondaryBtnText, primaryBtnText } = this.props;
```

Comment:
What do you think of adding 'actions' instead of 'secondaryBtnText' and 'primaryBtnText'? 
`const { children, title, actions } = this.props;`

Reply:
Are you thinking actions being an object with primary and secondary properties or an array?
actions = {primary: "Yes", secondary: "No" },  actions = ["Yes", "No"]

Reply:
Got it, sorry responded to the first comment without reading the others 😄 Going to change the control now to see how it goes